src/Common/src/System/Text/ValueStringBuilder.cs

```diff
@@ -0,0 +1,189 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Buffers;
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+
+namespace System.Text
+{
+    internal ref struct ValueStringBuilder
+    {
+        private char[] _arrayToReturnToPool;
+        private Span<char> _chars;
+        private int _pos;
+
+        public ValueStringBuilder(Span<char> initialBuffer)
+        {
+            _arrayToReturnToPool = null;
+            _chars = initialBuffer;
+            _pos = 0;
+        }
+
+        public string GetString()
```

Comment:
Why not ToString()? To avoid virtual dispatch?

Reply:
@jkotas, are virtual overrides supported on ref-like types?  I thought "no" (which is why I used a different name, and I didn't want to use "new" to avoid confusion), but a simple test shows it works.  Is that supported, or is it playing with fire?

Reply:
Yes, they are supported.

They are just always called non-virtually because of you cannot get a boxed copy.

Reply:
> Yes, they are supported.

Ok, good to know.  I'll just make it ToString then.